engine/lib/phx/src/render/mesh.rs

```diff
@@ -31,584 +40,809 @@ pub struct Computed {
-}
+    pub fn from_bytes(buf: &mut Bytes) -> Mesh {
+        let mut mesh = Mesh::new();
 
-#[no_mangle]
-pub extern "C" fn Mesh_AddTri(this: &mut Mesh, i1: i32, i2: i32, i3: i32) {
-    Mesh_AddIndex(this, i1);
-    Mesh_AddIndex(this, i2);
-    Mesh_AddIndex(this, i3);
-}
+        let vertex_count = buf.read_i32();
+        let index_count = buf.read_i32();
 
-#[no_mangle]
-pub extern "C" fn Mesh_AddVertex(
-    this: &mut Mesh,
-    px: f32,
-    py: f32,
-    pz: f32,
-    nx: f32,
-    ny: f32,
-    nz: f32,
-    u: f32,
-    v: f32,
-) {
-    this.vertex.push(Vertex {
-        p: Vec3::new(px, py, pz),
-        n: Vec3::new(nx, ny, nz),
-        uv: Vec2::new(u, v),
-    });
-    this.version += 1;
-}
+        mesh.reserve_vertex_data(vertex_count);
+        mesh.reserve_index_data(index_count);
 
-#[no_mangle]
-pub unsafe extern "C" fn Mesh_AddVertexRaw(this: &mut Mesh, vertex: *const Vertex) {
-    this.vertex.push(*vertex);
-    this.version += 1;
-}
+        {
+            let this = &mut *mesh.shared.as_mut();
 
-#[no_mangle]
-pub extern "C" fn Mesh_DrawBind(this: &mut Mesh) {
-    /* Release cached GL buffers if the mesh has changed since we built them. */
-    if this.vbo != 0 && this.version != this.versionBuffers {
-        glcheck!(gl::DeleteVertexArrays(1, &this.vao));
-        glcheck!(gl::DeleteBuffers(1, &this.vbo));
-        glcheck!(gl::DeleteBuffers(1, &this.ibo));
-        this.vao = 0;
-        this.vbo = 0;
-        this.ibo = 0;
-    }
-
-    /* Generate cached GL buffers for fast drawing. */
-    if this.vbo == 0 {
-        glcheck!(gl::GenBuffers(1, &mut this.vbo));
-        glcheck!(gl::GenBuffers(1, &mut this.ibo));
-        glcheck!(gl::BindBuffer(gl::ARRAY_BUFFER, this.vbo));
-        glcheck!(gl::BindBuffer(gl::ELEMENT_ARRAY_BUFFER, this.ibo));
-        glcheck!(gl::BufferData(
-            gl::ARRAY_BUFFER,
-            (this.vertex.len() as i32 as usize).wrapping_mul(std::mem::size_of::<Vertex>())
-                as gl::types::GLsizeiptr,
-            this.vertex.as_ptr() as *const _,
-            gl::STATIC_DRAW,
-        ));
+            this.vertex.resize(vertex_count as usize, Vertex::default());
+            this.index.resize(index_count as usize, 0);
 
-        /* TODO : 16-bit index optimization */
-        /* TODO : Check if 8-bit indices are supported by hardware. IIRC they
-         *        weren't last time I checked. */
+            buf.read(this.vertex.as_mut_slice());
+            buf.read(this.index.as_mut_slice());
+        }
 
-        glcheck!(gl::BufferData(
-            gl::ELEMENT_ARRAY_BUFFER,
-            (this.index.len() as i32 as usize).wrapping_mul(std::mem::size_of::<i32>())
-                as gl::types::GLsizeiptr,
-            this.index.as_ptr() as *const _,
-            gl::STATIC_DRAW,
-        ));
+        mesh
+    }
 
-        glcheck!(gl::GenVertexArrays(1, &mut this.vao));
-        glcheck!(gl::BindVertexArray(this.vao));
+    // #[bind(name = "FromSDF")]
```

Comment:
Is this still needed?

Reply:
Nope, I noticed this isn't used anywhere. We can use `SDF_FromMesh` instead. Will delete.